core/corehttp/gateway_handler.go

```diff
@@ -602,3 +623,42 @@ func getFilename(s string) string {
 	}
 	return gopath.Base(s)
 }
+
+func (i *gatewayHandler) searchUpTreeFor404(r *http.Request, parsedPath ipath.Path) (ipath.Resolved, error) {
+	filename404, err := preferred404Filename(r.Header["Accept"])
+	if err != nil {
+		return nil, err
+	}
+
+	pathComponents := strings.SplitAfter(parsedPath.String(), "/")
```

Comment:
SplitAfter is probably not the function you're looking for. When we re-join that, we'll end up with two slashes.